### PR TITLE
[webview_flutter_web] Add the credentialless iframe attribute.

### PR DESCRIPTION
**Pull request by uldall**
This PR makes it possible to specify whether the `<iframe>` should have the attribute `credentialless` set to true or false. 

* https://github.com/flutter/flutter/issues/151557

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by uldall** [comment]
Example usage:
```
    late final PlatformWebViewControllerCreationParams params;
    if (WebViewPlatform.instance is WebWebViewPlatform) {
      params = WebWebViewControllerCreationParams(
        iFrameCredentialless: true,
      );
    } else {
      params = const PlatformWebViewControllerCreationParams();
    }

    _webViewController = WebViewController.fromPlatformCreationParams(params);
```

**Comment by uldall** [comment]
Example of how it looks when the attribute is added:
image

**Comment by stuartmorgan** [comment]
Thanks for the contribution! You’ve checked boxes in the PR checklist above that are not reflected in this PR, so I’m assuming this is a work in progress and am marking it as a Draft. Please review the checklist, updating the PR as appropriate, and when the checklist reflects the state of the PR as posted please feel free to mark it as ready for review.

**Comment by uldall** [comment]
> Thanks for the contribution! You’ve checked boxes in the PR checklist above that are not reflected in this PR, so I’m assuming this is a work in progress and am marking it as a Draft. Please review the checklist, updating the PR as appropriate, and when the checklist reflects the state of the PR as posted please feel free to mark it as ready for review.

I have added tests and comments now. I am not sure about whether I am supposed to update the pubspec.yaml and CHANGELOG.md? Sorry about the potentially stupid question. This is my first PR on the Flutter project.

**Comment by uldall** [comment]
Would it be preferable if I changed this PR to use a Map<String, String> of attributes instead so it became more generic?

**Comment by stuartmorgan** [comment]
From triage: Ping @ditman on the design question above.

File: packages/webview_flutter/webview_flutter_web/lib/src/web_webview_controller.dart [hunk 1]
```diff
@@ -18,19 +18,32 @@ import 'http_request_factory.dart';
...
     @visibleForTesting this.httpRequestFactory = const HttpRequestFactory(),
-  }) : super();
+    bool iFrameCredentialless = false,
+  })  : iFrame = web.HTMLIFrameElement()
+          ..id = 'webView${_nextIFrameId++}'
+          ..style.width = '100%'
+          ..style.height = '100%'
+          ..style.border = 'none'
+          ..setAttribute(
+              'credentialless', iFrameCredentialless ? 'true' : 'false'),
```
Comment (ditman) [inline packages/webview_flutter/webview_flutter_web/lib/src/web_webview_controller.dart:32]: Setting `credentialless` to any value results it in being `true`

* https://jsfiddle.net/ditman/obufjvtk/

Set the attribute only `iFrameCredentialless` is `true`.

**Review by ditman — CHANGES_REQUESTED** [review]
Small nit with setting the `credentialless` attribute only when really needed.

**Comment by ditman** [comment]
> Would it be preferable if I changed this PR to use a Map<String, String> of attributes instead so it became more generic?

@uldall I think it's better that we do this using the type safety of Dart and `package:web` rather than a _stringly_ typed map :)

(PS: `credentialless` is only supported in Chrome browsers for now, we may need to add a note to the parameter itself so users are aware of this limitation?)

**Comment by stuartmorgan** [comment]
@uldall Are you still planning on updating this PR based on the feedback above?

**Comment by uldall** [comment]
I am not sure why GitHub closed my PR just because I updated my fork..

**Comment by stuartmorgan** [comment]
Since this is marked as a draft and hasn't been updated in several months, I’m going to close it so that our PR queue reflects active PRs. Please don't hesitate to submit a new PR if you decide to revisit this. Thanks!

**Pull request closed without merging**